app/questionnaire/router.py

```diff
@@ -321,12 +321,19 @@ def _get_return_to_for_grand_calculated_summary(
             ),
             routing_path,
         ):
+            return_to_answer_id = None
+            if return_location.return_to_answer_id:
+                return_to_answer_ids = return_location.return_to_answer_id.split(",")
+                return_to_answer_id = "".join(return_to_answer_ids.pop())
```

Comment:
because of the comment below, on the calculated summary page, you've ended up with two answer ids in the `return_to_answer_id` but when returning to a grand calculated summary, this currently should never be in a position where there are 2 - so I think the changes to this function won't be needed if the other one is fixed

Reply:
Looks like the suggestion below fixed it.